okhttp-tests/src/test/java/com/squareup/okhttp/URLConnectionTest.java

```diff
@@ -3212,6 +3212,25 @@ private void zeroLengthPayload(String method)
     }
   }
 
+  @Test public void urlRedirectToHostWithNul() throws Exception {
+    String redirectUrl = "http://host\u0000/";
+    server.enqueue(new MockResponse().setResponseCode(302)
+        .addHeaderLenient("Location", redirectUrl));
+
+    HttpURLConnection urlConnection = client.open(server.getUrl("/"));
+    assertEquals(302, urlConnection.getResponseCode());
+    assertEquals(redirectUrl, urlConnection.getHeaderField("Location"));
+  }
+
+  @Test public void urlHostWithNonLdhAscii() throws Exception {
```

Comment:
letter, decimal, hyphen. I'd not seen it before, but it is thrown around in IDN errors and unicode docs (e.g. https://tools.ietf.org/html/rfc5890) so I assumed I was just being ignorant. Happy to rename.